src/client/dfs/dfs.c

```diff
@@ -3306,6 +3312,42 @@ dfs_ostat(dfs_t *dfs, dfs_obj_t *obj, struct stat *stbuf)
 	return rc;
 }
 
+int
+dfs_statx(dfs_t *dfs, dfs_obj_t *parent, const char *name, struct stat *stbuf,
+	  daos_obj_id_t *expected_oid, int xnr, char *xnames[], void *xvals[],
+	  daos_size_t *xsizes)
+{
+	daos_handle_t	oh;
+	size_t		len;
+	int		rc;
+
+	if (dfs == NULL || !dfs->mounted)
+		return EINVAL;
+	if (parent == NULL)
+		parent = &dfs->root;
+	else if (!S_ISDIR(parent->mode))
+		return ENOTDIR;
+
+	if (name == NULL) {
+		if (strcmp(parent->name, "/") != 0) {
+			D_ERROR("Invalid path %s and entry name is NULL)\n",
+				parent->name);
+			return EINVAL;
+		}
+		name = parent->name;
+		len = strlen(parent->name);
+		oh = dfs->super_oh;
```

Comment:
dfs_statx() is an internal function so this code is essentially unreachable.  Same for L3326

Reply:
i disagree. i can add a test case in dfs_test and make this code alive :-)
anyway at some point there might be someone asking for a similar API and we might want to expose this.
but for now it depends on how you use it (even internally).